sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -661,7 +666,9 @@ class Analyzer(
           val evaluatedOrderings = resolvedAliasedOrdering.zip(sortOrder).map {
             case (evaluated, order) =>
               val index = originalAggExprs.indexWhere {
-                case Alias(child, _) => child semanticEquals evaluated.child
+                case a@Alias(child, _) =>
+                  (child semanticEquals evaluated.child) ||
+                    a.exprId == evaluated.child.asInstanceOf[AttributeReference].exprId
                 case other => other semanticEquals evaluated.child
```

Comment:
@cloud-fan **THANKS !!** Somehow i kept thinking that we need the full list of sort attributes for pushdown determination. After your comment , it makes sense as any resolved sort order attribute(s) must be resolved from the aggregate expressions and hence its ok to not considered. Right ?

One other question , after our a change a lot of cases are now going through this codepath and here we add an extra projection above the sort. Should we add this only when we have added at least one pushdown attribute ? Or should we fix the tescases instead ?